src/SDK/MonitoringApplianceEmulator/ViewModels/ChartPropertyControlViewModel.cs

```diff
@@ -0,0 +1,204 @@
+            }
+
+            switch (chartAlertProperty.XAxisType)
+            {
+                case ChartAxisType.Date:
+                    this.InitializeDateTimeXAxisChart(chartAlertProperty);
+                    break;
+                case ChartAxisType.Number:
+                    this.InitializeNumberXAxisChart(chartAlertProperty);
+                    break;
+                case ChartAxisType.String:
+                    throw new NotImplementedException($"Charts with {ChartAxisType.String} X axis type will be supported soon.");
+                default:
+                    throw new InvalidOperationException($"Charts with X axis type {chartAlertProperty.XAxisType} are not supported.");
+            }
+        }
+
+        /// <summary>
+        /// Gets the series collection.
+        /// </summary>
+        public SeriesCollection SeriesCollection
+        {
+            get
+            {
+                return this.seriesCollection;
+            }
+
+            private set
+            {
+                this.seriesCollection = value;
+                this.OnPropertyChanged();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the series collection.
+        /// </summary>
+        public Func<double, string> XAxisFormatter { get; set; }
+
+        /// <summary>
+        /// Gets the chart title.
+        /// </summary>
+        public string Title { get; }
+
+        /// <summary>
+        /// Converts a point's coordinate value to double.
+        /// </summary>
+        /// <param name="coordinateValue">The chart alert property that should be displayed.</param>
+        /// <param name="coordinateName">The coordinate name (e.g. X, Y).</param>
+        /// <returns>The coordinate value as double</returns>
+        private static double ConvertCordinateValueToDouble(object coordinateValue, string coordinateName)
+        {
+            try
+            {
+                return Convert.ToDouble(coordinateValue, CultureInfo.InvariantCulture);
+            }
+            catch (Exception e) when (e is FormatException || e is InvalidCastException || e is OverflowException)
+            {
+                throw new InvalidCastException($"The data point's {coordinateName} value - '{coordinateValue}', is not of a numeric type", e);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a chart with X axis from type <see cref="SmartDetectors.ChartAxisType.DateAxis"/>.
+        /// </summary>
+        /// <param name="chartAlertProperty">The chart alert property that should be displayed.</param>
+        private void InitializeDateTimeXAxisChart(ChartAlertProperty chartAlertProperty)
+        {
+            var chartValues = new ChartValues<ChartDataPoint<DateTime>>();
+            var sortedDateTimePoints = chartAlertProperty.DataPoints.OrderBy(point => point.X);
+            foreach (var dataPoint in sortedDateTimePoints)
```

Comment:
>foreach (var dataPoint in sortedDateTimePoints) [](start = 12, length = 47)

You can replace the entire loop with chartValues = sortedDateTimePoints.Select(...)